app/src/backend/config.ts

```diff
@@ -0,0 +1,101 @@
+interface Config {
+  [index: string]: string | boolean | object;
+  TOOL_PROJECT_TAG: string;
+  DATA_PROJECT_TAG: string;
+  DOWNLOADABLE_TAG: string;
+  NEEDS_ANALYSIS_TAG: string;
+  AUTOUPDATE_ENABLED: boolean;
+  CHROMIUM_MENU: boolean;
+  UPDATE_SERVER: string;
+  DOWNLOAD_INFO: {
+    [index: string]: object;
```

Comment:
Just because the compiler was complaining without them.

Reply:
This is because the usage in `dependency.getDownloadInfo` is unsafe. You cannot guarantee the return value from an object when indexing with arbitrary strings.

If you want type safety, my suggestion would be to 1) normalize `DOWNLOAD_INFO`:

```typescript
DOWNLOAD_INFO: {
  ANACONDA: {
    WIN32: {
      IA32: DXDownloadInfo;
      X64: DXDownloadInfo;
    };
    DARWIN: {
      IA32: null;
      X64: DXDownloadInfo;
    }
    LINUX: {
      IA32: DXDownloadInfo;
      X64: DXDownloadInfo;
    };
  };
};
```

And 2) use type unions to add constraints to package name, platform, and architecture:

```typescript
type Package = "ANACONDA";
type Platform = "DARWIN" | "LINUX" | "WIN32";
type Architecture = "IA32" | "X64";

const parsePlatform = (s: string): Platform => {
  switch (s) {
    case "darwin": return "DARWIN";
    case "linux": return "LINUX";
    case "win32": return "WIN32";
    default: throw new Error("invalid platform");
  }
}

const parseArchitecture = (s: string): Architecture => {
  switch (s) {
    case "ia32": return "IA32";
    case "x64": return "X64";
    default: throw new Error("invalid architecture");
  }
}

export const getDownloadInfo = (name: Package): DXDownloadInfo | null => {
  try {
    const platformUpper = parsePlatform(platform);
    const archUpper = parseArchitecture(arch);
    return config.DOWNLOAD_INFO[name][platformUpper][archUpper];
  } catch (_e) {
    return null;
  }
};
```